### PR TITLE
add dtypes to stat classes

### PR DESCRIPTION
**Pull request by ahnitz**
* Makes the stat class a bit more legible code wise
 * Allows some of the dtypes to be queried by higher level codes

### CODE REVIEW

**Comment by spxiwh** [comment]
seal_of_approval

**Comment by tdent** [comment]
Why was this assigned to me?  Please explain what this is about.

File: pycbc/events/stat.py [hunk 1]
```diff
@@ -50,6 +49,8 @@ def __init__(self, files):
             f = h5py.File(filename, 'r')
             stat = f.attrs['stat']
             self.files[stat] = f
+        self.single_dtype = numpy.float32
```
Comment (tdent) [inline pycbc/events/stat.py]: Please add a comment to explain what single_dtype is intended to do here or in general for Stat subclasses.

**Comment by ahnitz** [comment]
Add the following
* move phase td signal calculation into its own method
* add possible stub for a combined phase + exp fit class

**Comment by tdent** [comment]
ok, so I think you want to rebase this against master now.  note that I made get_newsnr a module-level function rather than a method since it doesn't in fact call the class instance, it's just a thin wrapper of the events.newsnr function.

**Comment by tdent** [comment]
Can you remove the 'stub' combined statistic (git stash or whatever), then it should be fine to merge.

**Pull request merged**